netflix_utils.py

```diff
@@ -0,0 +1,160 @@
+
+        #Global step(iteration) identifier. Will be incremented, not trained.
+        self.global_step_ = tf.get_variable('global_step', initializer=0, trainable=False)
+
+    def restore_existing_checkpoint(self, sess):
+        self.saver_ = tf.train.Saver() #To be able to save and restore variable status
+        # If a checkpoint was given in argument to the script, we will import the
+        # variables' status as it was at that checkpoint. NB : -1 implies latest checkpoint.
+        if self.args_.checkpoint is not None and os.path.exists(os.path.join(self.args_.logdir, 'checkpoint')):
+            if self.args_.checkpoint == -1:#latest checkpoint
+                self.saver_.restore(sess, tf.train.latest_checkpoint(self.args_.logdir))
+            else:#Specified checkpoint
+                self.saver_.restore(sess, os.path.join(self.args_.logdir, self.model_name_+".ckpt-"+str(self.args_.checkpoint)))
+            logging.debug('Model restored to step ' + str(self.global_step_.eval(sess)))
+
+    def setup_projector(self, movie_tensor_name):
+        ############################################################################
+        ## Visualisation configuration for TensorBoard
+
+        # Format: tensorflow/contrib/tensorboard/plugins/projector/projector_config.proto
+        config = projector.ProjectorConfig()
+
+        # You can add multiple embeddings. Here we add only one.
+        embedding = config.embeddings.add()
+        embedding.tensor_name = movie_tensor_name
+        # Link this tensor to its metadata file (e.g. labels).
+        embedding.metadata_path = '../movie_titles.tsv'
+
+        # Use the same LOG_DIR where you stored your checkpoint.
+        summary_writer = tf.summary.FileWriter(self.args_.logdir)
+
+        # The next line writes a projector_config.pbtxt in the logdir.
+        # TensorBoard will read this file during startup.
+        projector.visualize_embeddings(summary_writer, config)
+
+    def train_model(self, sess, train_step, accuracy, mse, loss,
+                    train_data_update_freq, test_data_update_freq,
+                    sess_save_freq, batch_size):
+
+        logging.debug('Loading netflix data')
+        data = netflix_data.DataSet.fromfile(self.args_.input)
+        train_batch_iter = data.iter_batch(batch_size)
+        test_data = netflix_data.DataSet.fromfile(self.args_.test_set)
+
+        logging.debug('Training model')
+        while self.global_step_.eval(sess) < self.args_.n_iter:
+            batch = next(train_batch_iter) # next batch of data to train on.
+
+            # the backpropagation training step
+            sess.run(train_step, feed_dict={self.user_ids_: batch.user_ids, self.movie_ids_: batch.movie_ids, self.ratings_: batch.ratings})
+            i = self.global_step_.eval(sess)
+
+            # compute training values for visualisation.
+            if i % train_data_update_freq == 0:
+                a, m, l = sess.run([accuracy, tf.sqrt(mse), loss],
```

Comment:
looked into a performance issue; we need to have
rmse = tf.sqrt(mse)
outside of the loop so it doesn't get created everytime